Tests/ASUICollectionViewTests.m

```diff
@@ -95,12 +104,13 @@ - (void)_testSupplementaryNodeAtIndexPath:(NSIndexPath *)indexPath sectionCount:
   cv.dataSource = dataSource;
   if (shouldFail) {
     XCTAssertThrowsSpecificNamed([cv layoutIfNeeded], NSException, NSInternalInconsistencyException);
-  } else {
-    [cv layoutIfNeeded];
-    XCTAssertEqualObjects(attr, [cv layoutAttributesForSupplementaryElementOfKind:@"SuppKind" atIndexPath:indexPath]);
-    XCTAssertEqual(view, [cv supplementaryViewForElementKind:@"SuppKind" atIndexPath:indexPath]);
+    // Early return because behavior after exception is thrown is undefined.
+    return;
   }
 
+  [cv layoutIfNeeded];
+  XCTAssertEqualObjects(attr, [cv layoutAttributesForSupplementaryElementOfKind:@"SuppKind" atIndexPath:indexPath]);
+  XCTAssertEqual(view, [cv supplementaryViewForElementKind:@"SuppKind" atIndexPath:indexPath]);
```

Comment:
Unfortunately, under the new OCMock, if an exception is thrown while inside a mock object method, and you later verify, it will fail and report that an exception was thrown. So we will have to avoid verifying after testing exception-throwing, which is sucky but not worth holding back on OCMock 2.2.